codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -314,21 +314,21 @@ protected String determineEncoding(String acceptEncoding) {
             }
         }
         if (brQ > 0.0f || zstdQ > 0.0f || gzipQ > 0.0f || deflateQ > 0.0f) {
-            if (brQ != -1.0f && brQ >= zstdQ) {
-                return Brotli.isAvailable() ? "br" : null;
-            } else if (zstdQ != -1.0f && zstdQ >= gzipQ) {
+            if (brQ != -1.0f && brQ >= zstdQ && this.brotliOptions != null) {
+                return "br";
+            } else if (zstdQ != -1.0f && zstdQ >= gzipQ && this.zstdOptions != null) {
                 return "zstd";
-            } else if (gzipQ != -1.0f && gzipQ >= deflateQ) {
+            } else if (gzipQ != -1.0f && gzipQ >= deflateQ && this.gzipOptions != null) {
                 return "gzip";
-            } else if (deflateQ != -1.0f) {
+            } else if (deflateQ != -1.0f && this.deflateOptions != null) {
                 return "deflate";
             }
         }
         if (starQ > 0.0f) {
-            if (brQ == -1.0f) {
-                return Brotli.isAvailable() ? "br" : null;
+            if (brQ == -1.0f && Brotli.isAvailable()) {
```

Comment:
you could make this a bit more consistent by using `&& this.brotliOptions != null`. Same is true for the change below.

Reply:
Indeed!